mteb/model_meta.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from datetime import date
+from functools import partial
+from typing import Any, Callable, Literal
+
+from pydantic import BaseModel, BeforeValidator, TypeAdapter
+from sentence_transformers import SentenceTransformer
+from typing_extensions import Annotated
+
+from mteb.encoder_interface import Encoder, EncoderWithQueryCorpusEncode
+
+from .languages import ISO_LANGUAGE_SCRIPT
+
+Frameworks = Literal["Sentence Transformers", "PyTorch"]
+
+pastdate_adapter = TypeAdapter(date)
+STR_DATE = Annotated[
+    str, BeforeValidator(lambda value: str(pastdate_adapter.validate_python(value)))
+]  # Allows the type to be a string, but ensures that the string is a valid date
+
+
+def sentence_transformers_loader(model_name: str, revision: str) -> SentenceTransformer:
+    return SentenceTransformer(model_name_or_path=model_name, revision=revision)
+
+
+class ModelMeta(BaseModel):
+    """The model metadata object.
+
+    Attributes:
+        loader: the function that loads the model. If None it will just default to loading the model using the sentence transformer library.
+        name: The name of the model, ideally the name on huggingface.
+        n_parameters: The number of parameters in the model. Can be None if the the number of parameters is not known (e.g. for proprietary models) or
+            if the loader returns a SentenceTransformer model from which it can be derived.
+        memory_usage: The amount of memory the model uses. Can be None if the memory usage is not known (e.g. for proprietary models).
+        max_tokens: The maximum number of tokens the model can handle. Can be None if the maximum number of tokens is not known (e.g. for proprietary
+            models).
+        embedding_dimension: The dimension of the embeddings produced by the model. Currently all models are assumed to produce fixed-size embeddings.
+        revision: The revision number of the model.
+        release_date: The date the model was released.
```

Comment:
Is this referring to the release date of this particular revision, or the first ever revision?
```suggestion
        release_date: The date the model's revision was released.
```